crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -690,9 +699,21 @@ where
             ) => {
                 debug_assert_eq!(&self.current_assignments, &[]);
                 self.visit_expr(value);
-                self.push_assignment(aug_assign.into());
-                self.visit_expr(target);
-                self.pop_assignment();
+
+                // See https://docs.python.org/3/library/ast.html#ast.AugAssign
+                if matches!(
+                    **target,
+                    ast::Expr::Attribute(_)
+                        | ast::Expr::Subscript(_)
+                        | ast::Expr::Starred(_)
```

Comment:
I don't think `Starred` expressions are legal in `AugAssign`s:

The parser permits them in `Assign`s:

```pycon
>>> import ast
>>> print(ast.dump(ast.parse('*y = z'), indent=2))
Module(
  body=[
    Assign(
      targets=[
        Starred(
          value=Name(id='y', ctx=Store()),
          ctx=Store())],
      value=Name(id='z', ctx=Load()))])
```

But not `AugAssign`s:

```pycon
>>> import ast
>>> print(ast.dump(ast.parse('*y += z'), indent=2))
Traceback (most recent call last):
  File "<python-input-4>", line 1, in <module>
    print(ast.dump(ast.parse('*y += z'), indent=2))
                   ~~~~~~~~~^^^^^^^^^^^
  File "/Users/alexw/.pyenv/versions/3.13.0/lib/python3.13/ast.py", line 54, in parse
    return compile(source, filename, mode, flags,
                   _feature_version=feature_version, optimize=optimize)
  File "<unknown>", line 1
    *y += z
    ^^
SyntaxError: 'starred' is an illegal expression for augmented assignment
```

Reply:
Ah, thanks. This means the Python docs are a bit misleading here. For `AugAssign`, they state that the *"target attribute cannot be of class [Tuple](https://docs.python.org/3/library/ast.html#ast.Tuple) or [List](https://docs.python.org/3/library/ast.html#ast.List), unlike the targets of [Assign](https://docs.python.org/3/library/ast.html#ast.Assign)."* — which I interpreted as: everything allowed for a normal assignment, but not `Tuple` or `List`.

Reply:
I agree the docs are bad here.

I think arguably the docs for the stdlib `ast` module say too _much_ here, though. They should instead say less, and link to the canonical docs for which expressions are allowed as augmented-assignment targets, which are at https://docs.python.org/3/reference/simple_stmts.html#augmented-assignment-statements (and which are much more precise)